apps/web/src/app/_components/search/search-filter.tsx

```diff
@@ -22,8 +24,24 @@ export default function SearchFilter() {
     },
   });
 
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const pathName = usePathname();
+
+  const createQueryString = useCallback(
+    (name: string, value: string) => {
+      const params = new URLSearchParams(searchParams);
+      params.set(name, value);
+      return params.toString();
+    },
+    [searchParams],
+  );
+
   function onSubmit(values: z.infer<typeof formSchema>) {
-    console.log(values);
+    // console.log(searchParams.get("search")); query the user search info
+    router.push(
+      pathName + `/?${createQueryString("search", values.searchText)}`,
+    );
   }
 
   return (
```

Comment:
I don't know if this is out of scope or just a stupid idea but should the search-filter component also match the search params if we enter it directly into the url? Like if we change the url manually to include `?search=klaviyo` should the search bar have "klaviyo" in the input section?

Reply:
hmm, this is a good point. i think different ticket (since on the front end and not a must-have)

Reply:
Yeah that is the plan for appending all the search and filter parameters to the search params. If we share a URL like `https://coopernu.vercel.app?search=klaviyo&workTerm=SUMMER&workEnvironment=HYBRID` with someone and they click on the link, then the search bar should have `klaviyo`, and the filter dropdowns for work term and work environment (TODO) should have the correct values.